File: src/Controllers/ItemController.php
<?php

namespace App\Controllers;

use App\Services\ItemService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

class ItemController {
    private ItemService $itemService;

    public function __construct() {
        $this->itemService = new ItemService();
    }

    public function getItems(Request $request, Response $response): Response {
        $items = $this->itemService->getAllItems();
        $data = array_map(fn($item) => [
            "id" => $item->getId(),
            "name" => $item->getName(),
            "status" => $item->getRentalDetails()
        ], $items);

        $response->getBody()->write(json_encode($data));
        return $response->withHeader('Content-Type', 'application/json');
    }

    public function rentItem(Request $request, Response $response): Response {
        $params = (array)$request->getParsedBody();
        try {
            $item = $this->itemService->rentItem($params['id'], $params['startDate'], $params['endDate']);
            $response->getBody()->write(json_encode(["message" => "Item rented successfully"]));
            return $response->withHeader('Content-Type', 'application/json');
        } catch (\Exception $e) {
            $response->getBody()->write(json_encode(["error" => $e->getMessage()]));
            return $response->withStatus(400)->withHeader('Content-Type', 'application/json');
        }
    }

    public function getItemDetails(Request $request, Response $response, array $args): Response {
        try {
            $item = $this->itemService->getItemById((int)$args['id']);
            $data = [
                "id" => $item->getId(),
                "name" => $item->getName(),
                "status" => $item->getRentalDetails()
            ];
            $response->getBody()->write(json_encode($data));
            return $response->withHeader('Content-Type', 'application/json');
        } catch (\Exception $e) {
            $response->getBody()->write(json_encode(["error" => $e->getMessage()]));
            return $response->withStatus(404)->withHeader('Content-Type', 'application/json');
        }
    }
}

File: controllers/item_controller.py
from flask import Blueprint, jsonify, request
from services.item_service import ItemService

item_bp = Blueprint('item_bp', __name__)
item_service = ItemService()

@item_bp.route('/', methods=['GET'])
def get_items():
    items = item_service.get_all_items()
    data = [{
        "id": item.id,
        "name": item.name,
        "status": item.get_rental_details()
    } for item in items]
    return jsonify(data), 200

@item_bp.route('/rent', methods=['POST'])
def rent_item():
    data = request.json
    try:
        item = item_service.rent_item(data['id'], data['startDate'], data['endDate'])
        return jsonify({
            "message": f"Item rented successfully from {data['startDate']} to {data['endDate']}"
        }), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 400

@item_bp.route('/<int:item_id>', methods=['GET'])
def get_item_details(item_id):
    try:
        item = item_service.get_item_by_id(item_id)
        return jsonify({
            "id": item.id,
            "name": item.name,
            "status": item.get_rental_details()
        }), 200
    except Exception as e:
        return jsonify({"error": str(e)}), 404
